data/trie/patriciaMerkleTrie_test.go

```diff
@@ -418,6 +427,42 @@ func TestPatriciaMerkleTrie_DeepCloneShouldWork(t *testing.T) {
 	assert.Equal(t, originalRoot, clonedTrie)
 }
 
+func TestPatriciaMerkleTrie_Rollback(t *testing.T) {
+	t.Parallel()
+	tr := initTrie()
+	_ = tr.Commit()
+	rootHash, _ := tr.Root()
+
+	_ = tr.Update([]byte("dog"), []byte("value of dog"))
+	_ = tr.Commit()
+
+	err := tr.Rollback(rootHash)
+	assert.Nil(t, err)
+
+	err = tr.Prune(rootHash)
+	assert.NotNil(t, err)
```

Comment:
should use assert.equal instead
You are also testing that it is not possible to remove from evictionList after rollback, maybe do another test instead.

Reply:
Done.